### DISCUSSION TITLE
No Space Available

### DISCUSSION THREAD

**Discussion by satviktechie1986** [hardkoded/puppeteer-sharp#2695] [Q&A]
Unknown encountered on server. Message:'System.IO.IOException: No space left on device : '/tmp/4rhxb1cg.23b'

   at System.IO.FileSystem.CreateDirectory(String fullPath)

   at System.IO.Directory.CreateDirectory(String path)

   at PuppeteerSharp.Helpers.TempDirectory..ctor(String path) in /home/runner/work/puppeteer-sharp/puppeteer-sharp/lib/PuppeteerSharp/Helpers/TempDirectory.cs:line 25

   at PuppeteerSharp.ChromiumLauncher.PrepareChromiumArgs(LaunchOptions options) in /home/runner/work/puppeteer-sharp/puppeteer-sharp/lib/PuppeteerSharp/ChromiumLauncher.cs:line 103

   at PuppeteerSharp.ChromiumLauncher..ctor(String executable, LaunchOptions options) in /home/runner/work/puppeteer-sharp/puppeteer-sharp/lib/PuppeteerSharp/ChromiumLauncher.cs:line 29

   at PuppeteerSharp.Launcher.LaunchAsync(LaunchOptions options) in /home/runner/work/puppeteer-sharp/puppeteer-sharp/lib/PuppeteerSharp/Launcher.cs:line 97
```[tasklist]
### Tasks


#############################################################################################


var bytes = ReadStream(responseStream);

//LambdaLogger.Log("step 2");
if (this.loggerFactory == null)
    this.loggerFactory = (ILoggerFactory)new LoggerFactory();
string[] args = {
    "--disable-dev-shm-usage",
    "--disable-background-timer-throttling",
    "--disable-extensions",
    "--no-sandbox",
    "--single-process"
};

var chromeLocation = new ChromiumExtractor(loggerFactory).ExtractChromium();


 var launchOptions = new LaunchOptions()
 {
     ExecutablePath = chromeLocation,
     Args = args,
     Headless = true 

 };


    using (var browser = await Puppeteer.LaunchAsync(launchOptions)) 
    using (var page = await browser.NewPageAsync())
    {
        try
        {
            page.DefaultTimeout = 3600;
            LambdaLogger.Log("step 4");

            var html = responseBody;

            PdfOptions objnav = new PdfOptions();
            objnav.Format = PaperFormat.A4;
            objnav.PrintBackground = true;
            objnav.MarginOptions.Bottom = "2cm";
            objnav.MarginOptions.Top = "2cm";
            objnav.MarginOptions.Left = "1cm";
            objnav.MarginOptions.Right = "1cm";

            if (!string.IsNullOrEmpty(report.FooterTemplate))
            {
                objnav.DisplayHeaderFooter = true;
                objnav.HeaderTemplate = "<div/>";
                objnav.FooterTemplate = report.FooterTemplate;
            }

            if (html.Contains("<visionplus>"))
            {
                await page.SetContentAsync(await ReplaceTags(html, report.DocumentData, report));
            }
            else
            {
                await page.SetContentAsync(await ReplaceTagsWithBM(html, report.DocumentData, report));
            }
            var result = await page.GetContentAsync();

            //LambdaLogger.Log("Waiting for Lambda Upload");

            var resultFile = report.IsCustomMargin ? await page.PdfStreamAsync() : await page.PdfStreamAsync(objnav);
            await UploadFileStreamSync(resultFile, report.BucketName, s3ObjectKey);
            //LambdaLogger.Log(s3ObjectKey);
        }
        catch (Exception ex)
        {
            LambdaLogger.Log("reportData " + report.DocumentData);
            Console.WriteLine("Error encountered  " + ex.Message);
            return string.Empty;
        }
        finally
        {
             
        }

    }
}

```[tasklist]
### Tasks
```

**Comment by kblok** [comment]
You are having an issue writing to disk

**Reply by satviktechie1986** [reply]
Hi Darío  , Thanks for  quick reply .
 
prior to using  "--single-process" args in launch options i was getting Below Error 

:'PuppeteerSharp.ProcessException: Failed to create connection
---> PuppeteerSharp.TargetClosedException: Protocol error(Target.setAutoAttach): Target closed. (The remote party closed the WebSocket connection without completing the close handshake.)

So i have added "--single-process" args in launch option to generate pdf , but now i think the temp directory is not getting cleared with this option so disk is getting fulled  and getting  error of no space left . i think the error  raised from TempDirectory.cs while creating the directory . 

I am using this function to generate pdf and saved to amazon S3 bucket .
Deployment  : AWS Lambda + .net 6 
PuppeteerSharp Version  : 10
HeadlessChromium.Puppeteer.Lambda.Dotnet Version : 1.1.0.84

Can you pleease guide ?
